src/server.js

```diff
@@ -178,7 +178,10 @@ module.exports = function (options = {}) {
 				performAuthentication.call(socket, {});
 			} else {
 				socket.on("auth:perform", performAuthentication);
-				socket.emit("auth:start", serverHash);
+				socket.emit("auth:start", {
+					serverHash: serverHash,
+					headerAuthEnabled: Helper.config.headerAuth.enable,
```

Comment:
I think this is another case where `public` would need to be checked as well.

I feel like we should really be erroring if `public` is set while ldap or header auth is enabled, since the latter settings will be totally ignored... then we could avoid this error-prone nonsense. But since that's a backwards incompatible change it'd probably be best to do that in a separate PR...